File: CLASS/Chapter6_practice/example6_15_viewflipper/src/main/java/com/example/example6_15_viewflipper/MainActivity.java
package com.example.example6_15_viewflipper;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ViewFlipper;

public class MainActivity extends AppCompatActivity {

    Button btnPrev, btnNext;
    ViewFlipper viewFlipper;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // region 위젯 연결
        btnPrev = (Button) findViewById(R.id.btnPrev);
        btnNext = (Button) findViewById(R.id.btnNext);
        viewFlipper = findViewById(R.id.viewFlipper);
        // endregion

        // region 이전화면 버튼 클릭 이벤트 처리
        btnPrev.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                viewFlipper.setFlipInterval(1000);
                viewFlipper.startFlipping();
            }
        });
        // endregion

        // region 다음화면 버튼 클릭 이벤트 처리
        btnNext.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                viewFlipper.stopFlipping();
            }
        });
        // endregion
    }
}

File: MyApplication8/myapplication/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*
import java.text.DateFormat
import java.text.SimpleDateFormat
import java.util.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var cal = Calendar.getInstance()

        cal.time = Date()

        var df:DateFormat = SimpleDateFormat("yyyy-MM-dd")
        println("current: ${df.format(cal.time)}")

        cal.add(Calendar.MONTH, 2)
        cal.add(Calendar.DATE, -3)
        println("after: ${df.format(cal.time)}")

        button.visibility = View.VISIBLE
    }
}
